src/rpc/fam_rpc_service_impl.cpp

```diff
@@ -132,23 +316,55 @@ Fam_Rpc_Service_Impl::signal_termination(::grpc::ServerContext *context,
     return ::grpc::Status::OK;
 }
 
+::grpc::Status
+Fam_Rpc_Service_Impl::reset_profile(::grpc::ServerContext *context,
+                                    const ::Fam_Request *request,
+                                    ::Fam_Response *response) {
+
+    RPC_SERVICE_PROFILE_INIT();
+    RPC_SERVICE_PROFILE_START_TIME();
+    fabric_reset_profile();
+    allocator->reset_profile();
+    return ::grpc::Status::OK;
+}
+
+void Fam_Rpc_Service_Impl::dump_profile() {
+    RPC_SERVICE_PROFILE_END();
+    fabric_finalize();
+    allocator->dump_profile();
+}
+
+::grpc::Status
+Fam_Rpc_Service_Impl::generate_profile(::grpc::ServerContext *context,
```

Comment:
what is the difference between dump_profile and this function? and wht are you sending signal to self? you can call teh function directly right?